### DISCUSSION TITLE
Quick Uniswap version detection

### DISCUSSION THREAD

**Discussion by AlexFigalex** [Elnaril/uniswap-universal-router-decoder#56] [Q&A]
Hi. I'm learning web3.py and came across your library, it's really cool work, thank you! I have a question a bit off-topic, but please answer if you can: if at the moment I buy a token, I only know its address. How can I quickly determine the version of its liquidity pool (uniswap v2 or uniswap v3) to generate a correct transaction? Thanks.

**Comment by Elnaril** [comment]
Hi!
You have several options; off the top of my head:
- Give the [uniswap smart path](https://github.com/Elnaril/uniswap-smart-path) a try: as a bonus you'll get the path to get the _best_ price! ;) 
- Call the V2 and V3 factories to check the existence of a pool with your token and weth or some stablecoins
- Request a quote from V2 and V3 for your token against weth or some stablecoins

Hope this helps! :)
Closing this discussion as off topic indeed.

**Comment by Petair13** [comment]
Hi
